Src/WindowsServer/WindowsServer.Shared/AppServicesHeartbeatTelemetryModule.cs

```diff
@@ -0,0 +1,150 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+
+    /// <summary>
+    /// Provides default values for the heartbeat feature of Application Insights that
+    /// are specific to Azure App Services (Web Apps, Functions, etc...).
+    /// </summary>
+    internal class AppServicesHeartbeatTelemetryModule : ITelemetryModule, IDisposable
+    {
+        internal const int SiteNameMaxLength = 256;
+        internal const int StampNameMaxLength = 256;
+        internal const int HostNameMaxLength = 256;
+
+        /// <summary>
+        /// Environment variables and the Application Insights heartbeat field names that accompany them.
+        /// </summary>
+        internal static readonly KeyValuePair<string, string>[] WebHeartbeatPropertyNameEnvVarMap = new KeyValuePair<string, string>[]
+        {
+            new KeyValuePair<string, string>("appSrv_SiteName", "WEBSITE_SITE_NAME"),
+            new KeyValuePair<string, string>("appSrv_wsStamp", "WEBSITE_HOME_STAMPNAME"),
+            new KeyValuePair<string, string>("appSrv_wsHost", "WEBSITE_HOSTNAME")
+        };
+
+        private static AppServicesHeartbeatTelemetryModule instance;
+        private object lockObject = new object();
+        private bool isInitialized = false;
```

Comment:
does this double initialization control matter for this module? There is no harm in initializing twice. Furthermore - this protection in module creates a false perception that nobody will call `Initialize` twice. In reality it may easily happen with re-creating of entire `TelemetryConfiguration`

Reply:
I don't need to protect it, no. I do need to know when it gets re-initialized but that's all.